### PR TITLE
Create view and model for indirect Sqw

### LINKED ISSUE #33678: Indirect make S(Q,omega) MVP
This is to make the S(Q, omega) tab in indirect data reduction into MVP and add tests

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

This PR is to make the Indirect S(Q, omega) tab MVP compliant. currently as it is part of the IndirectDataReduction interface It is required to inherit from the IndirectDataReductionTab object, this means that the algorithm running is still being done in the presenter, and tests for the presenter can not be written.

**To test:**

1. open the indirect data reduction interface.
2. go to the Sqw tab. 
3. Load _red data.
4. check that the preview plot has worked and ranges for Q and E are set to the range of the data.
8. check rebin in energy and change the Q and E ranges to be smaller that the data ranges.
6. Click the run button.
7. Check the sqw workspace generated and check that the data range is the same as set in the interface.
8. uncheck rebin in energy and run again. check the energy range in the output has not been rebinned.

Fixes #33678

*This does not require release notes* because It is all internal chagnes, it should not be noticed by users.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/IndirectSqw.cpp [hunk 1]
```diff
@@ -22,189 +22,90 @@ using MantidQt::API::BatchAlgorithmRunner;
...
+      m_view(std::make_unique<IndirectSqwView>(parent)) {
   setOutputPlotOptionsPresenter(
-      std::make_unique<IndirectPlotOptionsPresenter>(m_uiForm.ipoPlotOptions, PlotWidget::SpectraContour));
+      std::make_unique<IndirectPlotOptionsPresenter>(m_view->getPlotOptions(), PlotWidget::SpectraContour));
+  connectSignals();
+}
 
-  connect(m_uiForm.dsSampleInput, SIGNAL(dataReady(QString const &)), this, SLOT(handleDataReady(QString const &)));
-  connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(sqwAlgDone(bool)));
+IndirectSqw::~IndirectSqw() = default;
```
Comment (AnthonyLim23) [inline qt/scientific_interfaces/Indirect/IndirectSqw.cpp:39]: what is this line doing?

File: qt/scientific_interfaces/Indirect/IndirectSqw.cpp [hunk 2]
```diff
@@ -22,189 +22,90 @@ using MantidQt::API::BatchAlgorithmRunner;
...
-  if (!errorMessage.isEmpty()) {
-    showMessageBox(errorMessage);
-  } else {
-    plotRqwContour(dataName.toStdString());
-    setDefaultQAndEnergy();
+  if (m_view->validate()) {
+    m_model->setInputWorkspace(dataName.toStdString());
+    m_model->setEFixed(getInstrumentDetail("Efixed").toStdString());
+    plotRqwContour();
+    m_view->setDefaultQAndEnergy();
```
Comment (AnthonyLim23) [inline qt/scientific_interfaces/Indirect/IndirectSqw.cpp:78]: does this mean we no longer produce the warning if the data is of the wrong type?

File: qt/scientific_interfaces/Indirect/IndirectSqw.cpp [hunk 3]
```diff
@@ -22,189 +22,90 @@ using MantidQt::API::BatchAlgorithmRunner;
...
-  sampleLogAlg->setProperty("LogText", "NormalisedPolygon");
-
-  auto inputToAddSampleLogProps = std::make_unique<MantidQt::API::AlgorithmRuntimeProps>();
-  inputToAddSampleLogProps->setPropertyValue("Workspace", sqwWsName.toStdString());
-
-  m_batchAlgoRunner->addAlgorithm(sampleLogAlg, std::move(inputToAddSampleLogProps));
-
-  // Set the name of the result workspace for Python export
-  m_pythonExportWsName = sqwWsName.toStdString();
+  // m_model->setupAlgorithm();
```
Comment (AnthonyLim23) [inline qt/scientific_interfaces/Indirect/IndirectSqw.cpp:92]: is this supposed to be here?

File: qt/scientific_interfaces/Indirect/IndirectSqwModel.cpp [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+ */
+IndirectSqwModel::IndirectSqwModel() {}
+
+void IndirectSqwModel::setupRebinAlgorithm(MantidQt::API::BatchAlgorithmRunner *batchAlgoRunner) {
+  if (m_rebinInEnergy) {
+    std::string eRebinString = std::to_string(m_eLow) + "," + std::to_string(m_eWidth) + "," + std::to_string(m_eHigh);
+    auto const eRebinWsName = m_inputWorkspace.substr(0, m_inputWorkspace.length() - 4) + "_r";
```
Comment (AnthonyLim23) [inline qt/scientific_interfaces/Indirect/IndirectSqwModel.cpp:55]: can we have comment on why 4 is subtracted?

File: qt/scientific_interfaces/Indirect/test/IndirectSqwModelTest.h [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+  void test_setupRebinAlgorithm() {
+    MantidQt::API::BatchAlgorithmRunner batch;
+    m_model->setInputWorkspace("Workspace_name_red");
+    m_model->setEMin(-0.4);
+    m_model->setEWidth(0.005);
+    m_model->setEMax(0.4);
+    m_model->setRebinInEnergy(true);
+    m_model->setupRebinAlgorithm(&batch);
+    batch.executeBatch();
```
Comment (AnthonyLim23) [inline qt/scientific_interfaces/Indirect/test/IndirectSqwModelTest.h:54]: we dont seem to have any validation of the result here.

**Comment by AnthonyLim23** [comment]
the GUI works as expected. It seems the checks have frozen

**Review by AnthonyLim23 — DISMISSED** [review]
All works as expected

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by martyngigg** [comment]
Test failure is completed unrelated.

**Pull request merged**